lib/trino-filesystem/src/main/java/io/trino/filesystem/local/LocalOutputFile.java

```diff
@@ -58,6 +59,24 @@ public OutputStream create(AggregatedMemoryContext memoryContext)
         }
     }
 
+    @Override
+    public void createExclusive(byte[] data)
+            throws IOException
+    {
+        try {
+            Files.createDirectories(path.getParent());
+            Path tempPath = Files.createTempFile(path.getParent(), path.getFileName().toString(), ".tmp");
```

Comment:
It does not feel that it is always safe to have temporary files in the target directory

Reply:
What if I move it under java.io.tmpdir?

Reply:
Then move can not work if that is on different fielsystem